idrstream/idr_stream.py

```diff
@@ -310,6 +315,61 @@ def clear_batch(self):
 
         self.logger.info("Temporary batch files cleared")
 
+    def add_batch_object_outlines(
+        self, batch_single_cell_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        add object outlines to single cell data for current batch
+        necessary because we compile original single cell df with pycytominer
+        but pycytominer cannot append additional metadata (like object outlines)
+
+        Parameters
+        ----------
+        batch_single_cell_df : pd.DataFrame
+            original batch single cell df
+
+        Returns
+        -------
+        pd.DataFrame
+            new batch single cell df with object outlines appended
+        """
+
+        locations_save_path = pathlib.Path(f"{self.DP_project_path}/inputs/locations/")
+        new_batch_single_cell_df = []
+
+        # iterate over location files in order of plate map data in batch_single_cell_df
+        for DNA_image_path in batch_single_cell_df["Metadata_DNA"].unique():
+            # split single cell dataframe into image dataframes to find object outlines for that image
+            DNA_image_single_cell_df = batch_single_cell_df.loc[
+                batch_single_cell_df["Metadata_DNA"] == DNA_image_path
+            ].reset_index(drop=True)
+            DNA_image_plate = DNA_image_single_cell_df["Metadata_Plate"].unique()[0]
+            DNA_image_well = DNA_image_single_cell_df["Metadata_Well"].unique()[0]
+            DNA_image_site = DNA_image_single_cell_df["Metadata_Site"].unique()[0]
+
+            # load object outlines for a particular image
+            DNA_image_locations_path = pathlib.Path(
+                f"{locations_save_path}/{DNA_image_plate}/{DNA_image_well}-{DNA_image_site}-Nuclei.csv"
```

Comment:
again, will this always be Nuclei?

Reply:
I will not always be nuclei, but I think it keeping `Nuclei.csv` here makes it clear what needs to be updated in the future.